Mlem/Views/Tabs/Feeds/Components/PostFeedView.swift

```diff
@@ -0,0 +1,167 @@
+        didSet {
+            versionSafePostSort = postSortType
+        }
+    }
+    
+    // If versionSafePostSort is defined at init, the post tracker won't detect that and start loading until a fraction of a second after the view draws; if the tracker is also empty, this leads to noPostsView flashing for a fraction of a second. This masks that behavior.
+    @State var suppressNoPostsView: Bool = true
+
+    let showCommunity: Bool
+
+    @State var errorDetails: ErrorDetails?
+    
+    init(postSortType: Binding<PostSortType>, showCommunity: Bool) {
+        @Dependency(\.siteInformation) var siteInformation
+        
+        if let siteVersion = siteInformation.version, postSortType.wrappedValue.minimumVersion <= siteVersion {
+            self._versionSafePostSort = .init(wrappedValue: postSortType.wrappedValue)
+        }
+        
+        self._postSortType = postSortType
+        self.showCommunity = showCommunity
+    }
+    
+    var body: some View {
+        content
+            .animation(.easeOut(duration: 0.2), value: postTracker.items.isEmpty)
+            .onChange(of: showReadPosts) { newValue in
+                if newValue {
+                    Task { await postTracker.removeFilter(.read) }
+                } else {
+                    Task { await postTracker.addFilter(.read) }
+                }
+            }
+            .task(id: siteInformation.version) {
+                await setDefaultSortMode()
+            }
+            .task(id: versionSafePostSort) {
+                defer { suppressNoPostsView = false }
+                
+                if let versionSafePostSort {
+                    await postTracker.changeSortType(
+                        to: versionSafePostSort,
+                        forceRefresh: postTracker.items.isEmpty
+                    )
+                }
+            }
+            .toolbar {
+                if versionSafePostSort != nil {
+                    if postTracker.feedType != .saved {
+                        ToolbarItem(placement: .primaryAction) { sortMenu }
+                    }
+                    ToolbarItemGroup(placement: .secondaryAction) {
+                        ForEach(genEllipsisMenuFunctions()) { menuFunction in
+                            MenuButton(menuFunction: menuFunction, confirmDestructive: nil)
+                        }
+                        Menu {
+                            ForEach(genPostSizeSwitchingFunctions()) { menuFunction in
+                                MenuButton(menuFunction: menuFunction, confirmDestructive: nil)
+                            }
+                        } label: {
+                            Label("Post Size", systemImage: Icons.postSizeSetting)
+                        }
+                    }
+                }
+            }
+    }
+    
+    var content: some View {
+        LazyVStack(spacing: 0) {
+            if postTracker.items.isEmpty || versionSafePostSort == nil {
+                noPostsView()
```

Comment:
Thoughts on adding a little more padding here?

```suggestion
                noPostsView()
                    .padding(.top, 50)
```